content/05.main-text.md

```diff
@@ -255,6 +255,50 @@ We converted Hetionet v1.0 to HetMat format and uploaded the `hetionet-v1.0.hetm
 
 ### Computing DWPCs with matrix multiplication
 
+Prior to this study, we used two implementations for computing DWPCs.
+The first is a pure Python implementation available in the [`hetnetpy.pathtools.DWPC`](https://github.com/hetio/hetnetpy/blob/aa16e6a7092c039a6b175a73a35c006e53acee20/hetnetpy/pathtools.py#L8-L21) function [@hetio-dag].
+The second uses a Cypher query, prepared by [`hetnetpy.neo4j.construct_dwpc_query`](https://github.com/hetio/hetnetpy/blob/aa16e6a7092c039a6b175a73a35c006e53acee20/hetnetpy/neo4j.py#L363-L393), that is executed by the Neo4j database [@rephetio; @doi:10.15363/thinklab.d112].
+Both of these implementations require traversing all paths between the source and target node.
+Hence, they are computationally cumbersome despite optimizations [@doi:10.15363/thinklab.d187].
+
+An alternative approach counts paths by multiplying adjacency matrixes.
+However, this approach actually counts walks,
+since it includes sequences of edges that traverse a single node (i.e walk) or edge (i.e. trail) multiple times.
+When computing network-based features to quantify the relationship between a source and target node,
+we would like to exclude traversing duplicate nodes (i.e. paths, not walks nor trails) [@doi:10.15363/thinklab.d134].
+Therefore, we invented a suite of algorithms to compute true path counts and DWPCs using matrix multiplication. 
+
+TODO: Describe the suite of DWPC algorithms. From the categorize function, there are:
+
+- no_repeats
+- disjoint
+- short_repeat
+- long_repeat
+- BAAB
+- BABA
+- repeat_around
+- interior_complete_group
+- disjoint_groups
+- other
+
+Include information on what lengths of metapath we have completely solved this for. Mention testing against existing methods.
+
+Mention approximations. Also note independent approximation work at https://github.com/mmayers12/hetnet_ml
```

Comment:
@zietzm and @kkloste finally writing up your suite of DWPC methods.

@zietzm IIRC you may have created some long form documentation on the various DWPC methods based on metapath category. Do you remember if you've created any descriptions of the methodology here that we could use for the manuscript?

@zietzm what do you think is the best way for us to describe the suite of DWPC matrix multiplication methods? Formulas or just examples that link to the code implementation?

Reply:
> you may have created some long form documentation

I recall doing this, but unfortunately I can't seem to find it. I'm going to keep looking

> what do you think is the best way for us to describe the suite of DWPC matrix multiplication methods? Formulas or just examples that link to the code implementation?

Perhaps a supplemental section with formulae and brief descriptions?